base_classes/NXroot.nxdl.xml

```diff
@@ -46,13 +46,32 @@
     <attribute name="file_update_time" type="NX_DATE_TIME">
         <doc>Date and time of last file change at close</doc>
     </attribute>
-    <attribute name="NeXus_version">
+    <attribute name="NeXus_version" deprecated="NAPI is frozen.">
         <doc>
             Version of NeXus API used in writing the file.
 
-            Only used when the NAPI has written the file.
             Note that this is different from the version of the
-            base class or application definition version number.
+            base class or application definition version number.            
+        </doc>
+    </attribute>
+    <attribute name="NeXus_repository">
+        <doc>
+            A repository containing the application definitions
+            used for creating this file.
+            If the NeXus_release attribute contains a commit distance and hash,
+            this should refer to this repository.
+        </doc>
+    </attribute>
+    <attribute name="NeXus_release">
+        <doc>
+            Version of NeXus definitions used in writing the file.
+            This may either be a date based NeXus release (see 
+            https://github.com/nexusformat/definitions/releases) or a version tag 
+            with a commit distance and source control (e.g., git) hash of
+            the form `vYYYY.MM.post1.dev&lt;commit-distance&gt;.g&lt;git-hash&gt;`.
+            If the version contains a commit distance and hash the
```

Comment:
What's the definition of commit distance?

Reply:
I added a more detailed description of the `NeXus_release` attribute, incl. the definition of commit distance (i.e., the number of commits since the release indicated by `vYYYY.MM`). I am happy to change that again if something remains unclear.